src/ripple/app/rdb/backend/RelationalDBInterfaceSqlite.cpp

```diff
@@ -884,38 +1035,44 @@ RelationalDBInterfaceSqliteImp::getOldestAccountTxs(
             .first;
     }
 
-    /* else use shard databases */
-    AccountTxs ret;
-    AccountTxOptions opt = options;
-    int limit_used = 0;
-    iterateTransactionForward(
-        opt.minLedger ? seqToShardIndex(opt.minLedger)
-                      : std::optional<std::uint32_t>(),
-        [&](soci::session& session, std::uint32_t index) {
-            if (opt.maxLedger && index > seqToShardIndex(opt.maxLedger))
-                return false;
-            auto [r, total] = ripple::getOldestAccountTxs(
-                session, app_, ledgerMaster, opt, limit_used, j_);
-            ret.insert(ret.end(), r.begin(), r.end());
-            if (!total)
-                return false;
-            if (total > 0)
-            {
-                limit_used += total;
-                opt.offset = 0;
-            }
-            else
-            {
-                total = ~total;
-                if (opt.offset <= total)
+    /* else use shard databases, if available */
+    if (existsShardTransaction())
+    {
+        AccountTxs ret;
+        AccountTxOptions opt = options;
+        int limit_used = 0;
+        iterateTransactionForward(
+            opt.minLedger ? seqToShardIndex(opt.minLedger)
+                          : std::optional<std::uint32_t>(),
+            [&](soci::session& session, std::uint32_t index) {
+                if (opt.maxLedger && index > seqToShardIndex(opt.maxLedger))
+                    return false;
+                auto [r, total] = ripple::getOldestAccountTxs(
+                    session, app_, ledgerMaster, opt, limit_used, j_);
+                ret.insert(ret.end(), r.begin(), r.end());
+                if (!total)
+                    return false;
+                if (total > 0)
+                {
+                    limit_used += total;
                     opt.offset = 0;
+                }
                 else
-                    opt.offset -= total;
-            }
-            return true;
-        });
+                {
+                    total = ~total;
+                    if (opt.offset <= total)
+                        opt.offset = 0;
+                    else
+                        opt.offset -= total;
+                }
+                return true;
+            });
+
+        return ret;
+    }
 
-    return ret;
+    /* else return empty value */
```

Comment:
No change needed, but IMO comments like this actually detract from the code. It's easy to see that it returns an empty value, there's no need to point it out with a comment. The problem with comments like this is it adds "noise" to the code. We want comments to have a purpose.

I normally wouldn't point out a single instance of a comment like this, but there are many such comments in this file. I'm OK leaving them in (it's a minor nit), but it's something to keep in mind for future code.

Reply:
Did not removed such a comments because they finished if-else comment constructions:
/* if A do this */
...
/* else, if B, do that */
...
/* else return default */

Reply:
Right, the code is structured like this:
```
    /* if databases exists, use it */
    if (existsTransaction())
    /* else use shard databases, if available */
    if (existsShardTransaction())
    /* else return empty value */
    return {};
```
Those comments don't really serve a purpose. Like I said in the previous comment, I don't mind leaving them, but please keep it in mind for future code.